### PR TITLE
[WIP] Fixes showing up on testing with Optimization.jl

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
The aim is to get this branch passing all tests in OptimizationBase.jl

### CODE REVIEW

**Comment by gdalle** [comment]
Thanks! It's weird that my tests didn't catch that, I'll add some more for typing of `extras`

File: DifferentiationInterface/ext/DifferentiationInterfaceFiniteDiffExt/DifferentiationInterfaceFiniteDiffExt.jl [hunk 1]
```diff
@@ -9,6 +9,7 @@ using DifferentiationInterface:
     NoJacobianExtras,
     NoPullbackExtras,
     NoPushforwardExtras
```
Comment (gdalle) [inline DifferentiationInterface/ext/DifferentiationInterfaceFiniteDiffExt/DifferentiationInterfaceFiniteDiffExt.jl]:
```suggestion
    NoPushforwardExtras,
```

**Comment by Vaibhavdixit02** [comment]
```
ERROR: MethodError: no method matching prepare_pullback_aux(::OptimizationDIExt.var"#1#13"{…}, ::AutoEnzyme{…}, ::Vector{…}, ::Float64, ::DifferentiationInterface.PullbackFast)

Closest candidates are:
  prepare_pullback_aux(::Any, ::Any, ::Any, ::Any, ::Any, ::DifferentiationInterface.PullbackSlow)
   @ DifferentiationInterface ~/.julia/dev/DifferentiationInterface/DifferentiationInterface/src/first_order/pullback.jl:81
  prepare_pullback_aux(::Any, ::Any, ::Any, ::Any, ::DifferentiationInterface.PullbackSlow)
   @ DifferentiationInterface ~/.julia/dev/DifferentiationInterface/DifferentiationInterface/src/first_order/pullback.jl:75

Stacktrace:
 [1] prepare_pullback(f::Function, backend::AutoEnzyme{Nothing}, x::Vector{Float64}, dy::Float64)
   @ DifferentiationInterface ~/.julia/dev/DifferentiationInterface/DifferentiationInterface/src/first_order/pullback.jl:68
 [2] prepare_gradient(f::OptimizationDIExt.var"#1#13"{…}, backend::AutoEnzyme{…}, x::Vector{…})
   @ DifferentiationInterface ~/.julia/dev/DifferentiationInterface/DifferentiationInterface/src/first_order/gradient.jl:48
```

The current error with Enzyme

**Comment by gdalle** [comment]
Can you provide an MWE, even if it's with unregistered branches?
This is the typical error when a backend package is not loaded

**Comment by gdalle** [comment]
Gonna merge this cause it works, please open another issue or PR once you have a reproducible failure!

**Comment by gdalle** [comment]
> The current error with Enzyme

Oh I get it, it's because you're using `AutoEnzyme(mode=nothing)`, which is an option offered by ADTypes (see https://github.com/SciML/ADTypes.jl/issues/24) but not supported by DifferentiationInterface

**Comment by gdalle** [comment]
See https://github.com/gdalle/DifferentiationInterface.jl/issues/210 for tracking

**Comment by Vaibhavdixit02** [comment]
Ahh that makes sense!

**Comment by gdalle** [comment]
it's fixed on the main branch with #211

**Pull request merged**